### ISSUE TITLE
nextion display disconnected, Relays working fine

### ISSUE THREAD

**Issue by scuba-si** [Blackymas/NSPanel_HA_Blueprint#1095]
Hi All,

I'm hoping this is a user error :)

I have gone through the instructions, which I can follow well, except I have to manually download the .bin file and then upload using https://web.esphome.io/ as I do not have my home assistant on https. MACOS Safari / Chrome used.

The units (I have tried 2 nspanel EU) will communicate with HA once reflashed, but I cannot seem to get the screen tft to update away from the sonoff one, I am using the current (4 day old) version of the nspanel_eu.tft and I have it hosted on my HA in the www/ folder.

Here is the contents of my yaml that goes into ESPHOME:

`substitutions:

###### CHANGE ME START ######

  device_name: "nspanelblack3" 
  wifi_ssid: "REDACTED"
  wifi_password: "REDACTED"

  nextion_update_url: "http://192.168.1.239:8123/local/nspanel_eu.tft"
  #nextion_update_url: "https://github.com/Blackymas/NSPanel_HA_Blueprint/raw/main/custom_configuration/nspanel_blank.tft"
  #nextion_update_url: "https://raw.githubusercontent.com/Blackymas/NSPanel_HA_Blueprint/main/nspanel_us.tft" # URL to Github

###### USE THIS ONLY FOR YOUR FIRST TFT UPLOAD, ONCE IT WORKED, REMOVE THESE LINES ######
external_components:
  - source: github://pr#2956
    components: [nextion]
    refresh: 1h
################################################################################

##### DO NOT CHANGE ANYTHING! #####

packages:
  ##### download esphome code from Github
  remote_package:
    url: https://github.com/Blackymas/NSPanel_HA_Blueprint
    ref: main
    files: [nspanel_esphome.yaml]
    refresh: 300s

### If I keep the 'USE ONLY FOR YOUR FIRST TFT UPLOAD' section I get an error when creating the bin file:

`INFO ESPHome 2023.8.3
INFO Reading configuration /config/esphome/nspanelblack3.yaml...
INFO Detected timezone 'Europe/London'
WARNING GPIO4 is a Strapping PIN and should be avoided.
Attaching external pullup/down resistors to strapping pins can cause unexpected failures.
See https://esphome.io/guides/faq.html#why-am-i-getting-a-warning-about-strapping-pins
**Failed config**

display.nextion: [source /config/esphome/.esphome/packages/566b10a3/nspanel_esphome.yaml:1666]
  id: disp1
  platform: nextion
  uart_id: tf_uart
  tft_url: http://192.168.1.239:8123/local/nspanel_eu.tft
  
  **[on_page] is an invalid option for [display.nextion]. Did you mean [on_wake]?**
  on_page: 
    - lambda: ESP_LOGW("display.disp1", "NEXTION PAGE CHANGED");
  on_setup: 
    then: 
      - lambda: |-
          id(disp1).goto_page("boot");
          id(disp1).set_component_text_printf("boot.esph_version", "%s", "4.0"); // ### esphome-version ###
          id(disp1).show_component("bt_reboot");
          id(timer_reset_all).execute("boot");
      - wait_until: api.connected
      - lambda: |-
          id(disp1).set_component_text_printf("boot.ip_addr", "%s", id(ip_address).state.c_str());
          auto ha_event = new esphome::api::CustomAPIDevice();
          ha_event->fire_homeassistant_event("esphome.nspanel_ha_blueprint",
            {
              {"type", "boot"},
              {"step", "start"}
            });
      - delay: 1s
      - lambda: |-
          // Set dimming values
          id(display_brightness).publish_state(id(display_brightness_global));
          id(display_dim_brightness).publish_state(id(display_dim_brightness_global));
          id(disp1).send_command_printf("brightness=%i", id(display_brightness_global));
          id(disp1).send_command_printf("settings.brightslider.val=%i", id(display_brightness_global));
          id(disp1).send_command_printf("brightness_dim=%i", id(display_dim_brightness_global));
          id(disp1).send_command_printf("settings.dimslider.val=%i", id(display_dim_brightness_global));
          id(nextion_init).publish_state(true);
          auto ha_event = new esphome::api::CustomAPIDevice();
          ha_event->fire_homeassistant_event("esphome.nspanel_ha_blueprint",
            {
              {"type", "boot"},
              {"step", "nextion_init"}
            });
          id(home_relay1_icon) = "\uE3A5";
          id(home_relay1_icon) = "\uE3A8";
          id(timer_reset_all).execute("boot");
      - lambda: |-
          id(disp1).send_command_printf("is_notification=0");
          id(notification_label).publish_state("");
          id(notification_text).publish_state("");
      - switch.turn_off: notification_unread
      - logger.log: Nextion start - Done!
`

So in the end I have proceeded without that section and got myself to where I am now, _Nextion Display Disconnected_.  Can anyone help?? 

Should I be looking at tasmota?

Under Diagnostics for the device I have the following:

Diagnostic

API uptime 11 minutes ago
Device uptime 11 minutes ago
**Nextion display Disconnected**
RSSI -53 dBm
Status Connected

**Comment by edwardtfn** [comment]
Can you download the file to your computer when using this url? 

http://192.168.1.239:8123/local/nspanel_eu.tft

**Comment by edwardtfn** [comment]
And is the file the right size?

**Comment by edwardtfn** [comment]
You have tried pr#2956... I would recommend the one from our wiki instead: https://github.com/Blackymas/NSPanel_HA_Blueprint/wiki/(EN)-Common-Issues-TFT-Upload#10-use-an-older-nextion-library-for-esphome

If that isn't working, there are lots of other tips in this page.

Please let me know if you progress...

**Comment by scuba-si** [comment]
> You have tried pr#2956... I would recommend the one from our wiki instead: https://github.com/Blackymas/NSPanel_HA_Blueprint/wiki/(EN)-Common-Issues-TFT-Upload#10-use-an-older-nextion-library-for-esphome
> 
> If that isn't working, there are lots of other tips in this page.
> 
> Please let me know if you progress...

Thank you, the file was the correct size.  Whilst waiting I played around with tasmota, and have now got as far as the QR code on the screen, so I think I'm back on track and currently trying to follow your comments in another thread for getting something on the screen before my wife kills me.

For a newb / non-coder I found the tasmota route was way easier:  

https://github.com/Blackymas/NSPanel_HA_Blueprint/issues/685#issuecomment-1524787174

**Comment by edwardtfn** [comment]
> have now got as far as the QR code on the screen

Great! So those pr# and exit reparse are not needed anymore... Just upload the final TFT.

**Comment by scuba-si** [comment]
Ha, well I think I just bricked it. System Data Error after attempting to upload the larger tft.

**Comment by edwardtfn** [comment]
I don't think it is bricked.
Try to power cycle, or just flash it again... Transfer `nspanel_blank.tft`...
It normally gets it back to life.
I got that message myself a couple of time, as I'm always installing different things on my tests, and my panels are all working fine now.

**Comment by DiggaTS** [comment]
Hey,

i have the same problemes with 4 nspanels...

`[13:12:22][D][nextion_upload:302]: Heap Size 61296, Bytes left 6050112
[13:12:22][D][nextion_upload:048]: Requesting range: bytes=1392640-1458175
WARNING nspanelpaul.local: Connection error occurred: [Errno 104] Connection reset by peer
INFO Processing unexpected disconnect from ESPHome API for nspanelpaul.local
WARNING Disconnected from API`

please help ;-)

**Comment by edwardtfn** [comment]
Another thing you can try is to revert your ESPHome back to v3.4.1 and try to transfer the TFT before upgrading ESPHome to v4.0. Looks like that code was taking a bit less memory...
To revert to v3.4.1, change in your yaml, on the `remote_package` area, from `ref: main` to `ref: v3.4.1`.

**Comment by DiggaTS** [comment]
Argh ... Same Problem. I have also tryed to downgrade to 3.4.1 (ESP and TFT) After a failed upgrade try - same error but @26% and not 16%!

I will try to make a new device in esphome. I think there are to less memory on this device

**Comment by edwardtfn** [comment]
But have you tried that latest code from #1117?

**Comment by DiggaTS** [comment]
I have tryed to upload the tft 3 times... and now it works ... it stops by 80% 2 times ... 
It is not good but it is a workaround. ;-)

Thanks for the support and the better NSPanel in home assistant!
 [1117](https://github.com/Blackymas/NSPanel_HA_Blueprint/discussions/1117)

**Comment by deejaybeam** [comment]
@DiggaTS have you done this with the new nspanel_esphome_upload_tft.yaml file?

**Comment by DiggaTS** [comment]
Yes i have done this with the code sniped from the last post from 1117

**Comment by edwardtfn** [comment]
@scuba-si, is this still an issue with the latest release or should we close this?
